conda/models/record.py

```diff
@@ -75,3 +75,4 @@ class Record(DictSafeMixin, ImmutableEntity):
     link = ComposableField(Link, required=False)
 
     with_features_depends = MapField(required=False)
+    preferred_env = StringField(required=False, nullable=True)
```

Comment:
oh, this `required=False, nullable=True` should mean that `hasattr(record, 'preferred_env')` is always `True`.